api/query/cache/index/filter.go

```diff
@@ -324,6 +324,18 @@ func (tl TestLabel) Filter(t TestID) bool {
 	}
 
 	labels, ok := tl.metadata[name]
+	dir := filepath.Dir(name)
+	// Dir terminates with either '.' (when the top-level is a file) or '/'
+	// (when the top-level is a directory).
+	for !ok && len(dir) > 1 {
```

Comment:
On a second look, I want to point out a drawback of this for loop implementation. Since a test can have labels on a test level and directory level, this for loop will terminate as soon as `A` set of labels are found.  However, if a test `/a/b/c` is labelled to `foo` and `/a/*` is labelled to `bar`,  `label:bar` will not return the intended result, due to early termination at `for !ok ...`. This drawback will affect all wildcards implementation in this file.

Reply:
I will file a bug

Reply:
https://github.com/web-platform-tests/wpt.fyi/issues/3448